tests/test_controller.py

```diff
@@ -297,6 +297,53 @@ def test_libraries(self):
         self._is_library(ct, libraries[0])
         self._is_library(ks, libraries[1])
         self._is_library(nypl, libraries[2])
+    
+    def test_libraries_opds(self):
+        library = self._library(
+            name="Test Cancelled Library",
+            short_name="test_cancelled_lib",
+            library_stage=Library.CANCELLED_STAGE,
+            registry_stage=Library.TESTING_STAGE
+        )
+        response = self.controller.libraries()
+        libraries = response.get("libraries")
+
+        # There are currently four libraries,
+        # These libraries have been added from the previous test.
```

Comment:
They're added to every test by data_setup(), which is called by setup(), not left over from the previous test. Tests run in a subtransaction which is rolled back at the end of the test, so they're always separated.

Reply:
Ah got it.